src/main/java/org/eclipsefoundation/git/eca/resource/ValidationResource.java

```diff
@@ -300,6 +293,41 @@ private boolean isCommitter(ValidationResponse r, EclipseUser user, String hash,
 		return false;
 	}
 
+	/**
+	 * Retrieves projects valid for the current request, or an empty list if no data
+	 * or matching project repos could be found.
+	 * 
+	 * @param req the current request
+	 * @return list of matching projects for the current request, or an empty list
+	 *         if none found.
+	 */
+	private List<Project> retrieveProjectsForRequest(ValidationRequest req) {
+		String repoUrl = req.getRepoUrl();
+		// check for all projects that make use of the given repo
+		@SuppressWarnings("unchecked")
+		Optional<List<Project>> cachedProjects = cache.get("projects", () -> projects.getProject(),
+				(Class<List<Project>>) (Object) List.class);
```

Comment:
????

Reply:
If you are wondering about the double cast, that's because you can't directly reference a typed generic for casting. In order to get the proper typed generic, casting shenanigans were needed.